components/tyk-dashboard/templates/_helpers.tpl

```diff
@@ -51,8 +51,44 @@ http
 {{ include "tyk-dashboard.dash_proto" . }}://dashboard-svc-{{.Release.Name}}-tyk-dashboard.{{ .Release.Namespace }}.svc:{{ .Values.global.servicePorts.dashboard }}
 {{- end -}}
 
+{{/*
+    It lists all services in the release namespace and find a service
+    for Tyk Gateway with its label.
+*/}}
+{{- define "tyk-dashboard.gatewaySvcName" -}}
+    {{- $svcName := "" -}}
+    {{- $svcControl := "" -}}
+    {{- $svcGw := "" -}}
+   {{- $services := (lookup "v1" "Service" .Release.Namespace "") -}}
+   {{- if $services -}}
+       {{- range $index, $svc := $services.items -}}
+           {{- range $key, $val := $svc.metadata.labels -}}
+               {{- if and (eq $key "app") (contains "gateway-svc-" $val) -}}
+               {{- $svcGw = $svc.metadata.name | trim -}}
+               {{- else if and (eq $key "app") (contains "gateway-control-svc-" $val) -}}
+               {{- $svcControl = $svc.metadata.name | trim -}}
+               {{- end -}}
+           {{- end -}}
+       {{- end -}}
+   {{- end -}}
+   {{- if $svcControl -}}
+    {{- $svcName = $svcControl | trim -}}
+   {{- else -}}
+    {{- $svcName = $svcGw | trim -}}
+   {{- end -}}
+{{- $svcName | trim -}}
+{{- end }}
+
 {{- define "tyk-dashboard.gateway_host" -}}
-gateway-svc-{{.Release.Name}}-tyk-gateway.{{ .Release.Namespace }}.svc
+{{ if (include "tyk-dashboard.gatewaySvcName" .) }}
+{{- include "tyk-dashboard.gw_proto" . -}}://{{- include "tyk-dashboard.gatewaySvcName" . -}}.{{ .Release.Namespace }}.svc
```

Comment:
Probably we don't need protocol here since we are trying to gateway host?

Reply:
Updated in https://github.com/TykTechnologies/tyk-charts/pull/170/commits/45e79badc6324ba07735e46c233bd1c2bc6ac7e0. Can you please check again?